pkg/controller/agent/pod.go

```diff
@@ -209,15 +212,23 @@ func amendBuilderForFleetMode(params Params, fleetCerts *certificates.Certificat
 	}
 
 	if params.Agent.Spec.FleetServerEnabled {
-		// ECK creates CA and a certificate for Fleet Server to use. This volume contains those.
-		builder = builder.WithVolumeLikes(
-			volume.NewSecretVolumeWithMountPath(
-				fleetCerts.Name,
-				FleetCertsVolumeName,
-				FleetCertsMountPath,
-			))
-
 		builder = builder.WithPorts([]corev1.ContainerPort{{Name: params.Agent.Spec.HTTP.Protocol(), ContainerPort: FleetServerPort, Protocol: corev1.ProtocolTCP}})
+
+		// Only add certificate volumes if TLS is enabled.
+		if params.Agent.Spec.HTTP.TLS.Enabled() {
+			// ECK creates CA and a certificate for Fleet Server to use. This volume contains those.
+			builder = builder.WithVolumeLikes(
+				volume.NewSecretVolumeWithMountPath(
+					fleetCerts.Name,
+					FleetCertsVolumeName,
+					FleetCertsMountPath,
+				))
+		} else {
+			// Force FLEET_SERVER_HOST environment variable to Pod IP, as without this, Fleet Server only binds to localhost.
+			builder = builder.WithEnv(corev1.EnvVar{Name: "FLEET_SERVER_HOST", Value: "", ValueFrom: &corev1.EnvVarSource{
```

Comment:
1. Yes it should. Fixed.
3. Absolutely.  I've moved that into the function.

2.  Longer answer.  Let me test this one additional time.  IIRC, everything worked up until I started trying to use an actual agent attached to fleet.  Let me re-verify.

Reply:
After testing 3: It turns out that environment variable isn't needed after all.  I've removed it, and updated our agent tests to use actual URLs in the tests instead of `"url"` or similar.